src/rust/engine/src/externs/mod.rs

```diff
@@ -140,15 +140,16 @@ pub fn collect_iterable(value: &PyObject) -> Result<Vec<PyObject>, String> {
   }
 }
 
-pub fn getattr_from_frozendict(value: &PyObject, field: &str) -> BTreeMap<String, String> {
+/// Read a `FrozenDict[str, str]`.
+pub fn getattr_from_str_frozendict(value: &PyObject, field: &str) -> BTreeMap<String, String> {
   let frozendict = getattr(value, field).unwrap();
   let pydict: PyDict = getattr(&frozendict, "_data").unwrap();
   let gil = Python::acquire_gil();
```

Comment:
Maybe assert on the types of the key and value before unpacking?

Reply:
The extract() already will do that and raise TypeError if it's not a string